File: forge/src/main/java/net/shiroha233/roadweaver/persistence/forge/WorldDataProviderImpl.java
package net.shiroha233.roadweaver.persistence.forge;

import net.shiroha233.roadweaver.persistence.WorldDataProvider;

/**
 * Architectury @ExpectPlatform 实现类（Forge）
 */
public final class WorldDataProviderImpl {
    private static final WorldDataProvider INSTANCE = new ForgeWorldDataProvider();

    public static WorldDataProvider getInstance() {
        return INSTANCE;
    }
}

File: forge/src/main/kotlin/net/shiroha233/roadweaver/persistence/forge/WorldDataProviderImpl.kt
package net.shiroha233.roadweaver.persistence.forge

import net.shiroha233.roadweaver.persistence.WorldDataProvider

/**
 * Architectury @ExpectPlatform 的 NeoForge 端实现入口。
 * 提供 Common 抽象的实例。
 */
object WorldDataProviderImpl {
    private val INSTANCE: WorldDataProvider = ForgeWorldDataProvider()

    @JvmStatic
    fun getInstance(): WorldDataProvider {
        return INSTANCE
    }
}

@Suppress("MemberVisibilityCanBePrivate")
object WorldDataProviderAccessImpl {
    @JvmStatic
    fun getInstance(): WorldDataProvider = WorldDataProviderImpl.getInstance()
}
